### PR TITLE
Profile details page integration

### LINKED ISSUE #35: IN: Profile details page
Make profile details page dynamic based on user

### PR DESCRIPTION
**Pull request by MilesDev91**
### What this PR does:
- Hooks the profile details page to the backend.
- Profile is retrieved based on id passed in params /profile/:id
- Requests can be sent via the request form.

### Screenshots / Video:
image
image
image

### Any information needed to test this feature:
- Sign up as a new user to populate the necessary fields. Some can be edited in the profile settings page, others are not implemented and must be changed directly in the database or left as empty strings.
- Make sure you have a valid Profile to retrieve, in it you will want: name, photo (url), location, description, aboutMe, isSitter, and payRate. Photo is the most optional one (as it has a placeholder), the others will just default to empty strings. isSitter must be true or the page won't load the profile at all.

- Enter in a valid date range into the request form and submit. You should see a message that it was successful. Check your server to ensure successful creation.

### Any issues with the current functionality:
- The extra images are not currently implemented in the model, thus mock images will be used until they are there and uploadable. It should be a very easy integration once it is done.
- Not all values can be set in the settings, so naturally creating and editing a profile will not show all available information (although you can still send requests).
- Natural navigation to the profile details is not yet implemented. We currently must enter the profile id manually into the url.

closes #35

### CODE REVIEW

**Comment by MilesDev91** [comment]
I had to merge Bookings integration into this branch as there was a ton of Request related code I needed access to to complete this ticket.

File: server/controllers/profile.js
Comment (sundayezeilo) [inline server/controllers/profile.js:45]: :+1:

File: server/controllers/request.js
Comment (sundayezeilo) [inline server/controllers/request.js]: Good documentation

File: server/controllers/request.js
Comment (sundayezeilo) [inline server/controllers/request.js]: A failure message should also be returned to the client on failure

File: server/controllers/request.js
Comment (sundayezeilo) [inline server/controllers/request.js]: :+1:

**Review by sundayezeilo — COMMENTED** [review]
Nice integration :+1:

File: client/src/pages/Bookings/Bookings.tsx
Comment (rajivtitus) [inline client/src/pages/Bookings/Bookings.tsx:26]: Please remove logs to the console

Comment (MilesDev91) [reply client/src/pages/Bookings/Bookings.tsx:26]: Woops

**Review by rajivtitus — COMMENTED** [review]
No major comments for this PR (I may have a couple of minor ones on second review)
Let's make the minor changes in Profile Details UI and update this PR with the changes and also with the latest remote main 👍🏽

File: client/src/pages/ProfileDetails/ProfileDetails.tsx
Comment (rajivtitus) [inline client/src/pages/ProfileDetails/ProfileDetails.tsx:31]: Let's update the Snackbar here in case we fail to fetch the sitter's profile

File: server/controllers/request.js
Comment (rajivtitus) [inline server/controllers/request.js:60]: 👍🏽

File: client/src/pages/ProfileDetails/ProfileDetails.tsx
Comment (rajivtitus) [inline client/src/pages/ProfileDetails/ProfileDetails.tsx]: Why don't we consider rendering the entire Profile Details card only when a profile exists? In no profile is found we can just render large text - 'No Profile Found' (rather than displaying the cover photo and profile card) 😄

**Review by rajivtitus — APPROVED** [review]
Nice work! Just a comment to address before merging 👍🏽

File: client/src/pages/ProfileDetails/ProfileDetails.tsx
Comment (MilesDev91) [reply client/src/pages/ProfileDetails/ProfileDetails.tsx]: That is much better than what I did. Thanks

**Pull request merged**
